operator/src/main/java/oracle/kubernetes/operator/steps/DeleteDomainStep.java

```diff
@@ -34,14 +36,32 @@ public DeleteDomainStep(String namespace, String domainUID) {
 
   @Override
   public NextAction apply(Packet packet) {
-    return doNext(
-        Step.chain(
-            deleteIngresses(),
-            deleteServices(),
-            deletePods(),
-            deletePersistentVolumes(),
-            deletePersistentVolumeClaims()),
-        packet);
+    return doNext(Step.chain(deleteResources()), packet);
+  }
+
+  private Step[] deleteResources() {
+    ArrayList<Step> resources = new ArrayList<>();
```

Comment:
Have type be the generic List<Step> resources = new ArrayList<>();

Reply:
Do you mean using 
List resources = new ArrayList<>();
?
We use ArrayList<Step> because the result of this method is used as argument to Step.chain() which expects an array of Step, but I might be misunderstanding what you are suggesting though.

Reply:
@rjeberhard ?